### PR TITLE
sl-71: Start stepper target audience

### PR DESCRIPTION
**Pull request by Alexandr-mas**
image
step looks like this
image
Now, with this commit, this step is not called anywhere in the way that more will be created and then added to one array.

### CODE REVIEW

File: frontend/src/libs/components/stepper/step-target/stepper-target.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { useForm } from 'react-hook-form';
+
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+
+import { SelectTarget } from '../../select-target/select.js';
```
Comment (andanf-e) [inline frontend/src/libs/components/stepper/step-target/stepper-target.tsx]: Should use default select with enhanced stuff
```suggestion
import { Select } from '~/libs/components/components.js';
```

File: frontend/src/libs/components/select-target/styles.ts
Comment (andanf-e) [inline frontend/src/libs/components/select-target/styles.ts:1]: custom styled checkbox should be added into custom options component for multiselect

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/select-target/styles.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import { type StylesConfig } from 'react-select';
+import ceckIcon from 'src/assets/img/icons/choice.svg';
```
Comment (Alyona-Pisotska) [inline frontend/src/libs/components/select-target/styles.ts]: grammatical error?

File: frontend/src/libs/components/stepper/step-target/style.module.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.step-question {
+  margin: 23px 0 8px;
+  font-weight: 600;
+  font-size: 24px;
+  font-family: Roboto, sans-serif;
```
Comment (Alyona-Pisotska) [inline frontend/src/libs/components/stepper/step-target/style.module.scss]: You don`t need to add font-family - it is global

File: frontend/src/libs/components/stepper/step-target/style.module.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  line-height: 32px;
+  letter-spacing: 0;
+  text-align: left;
+}
+
+.chose-age {
+  margin: 0 0 44px;
+  font-weight: 400;
+  font-size: 14px;
+  font-family: Roboto, sans-serif;
```
Comment (Alyona-Pisotska) [inline frontend/src/libs/components/stepper/step-target/style.module.scss]: same

**Review by Alyona-Pisotska — CHANGES_REQUESTED** [review]

**Comment by Alexandr-mas** [comment]
image
step look like this

File: frontend/src/libs/components/stepper/step-target/style.module.scss [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+input[type="checkbox"]:checked::before {
+  display: inline-block;
+  width: 13px;
+  height: 13px;
+  color: #ffb61d;
+  font-size: 10px;
+  text-align: center;
+  vertical-align: middle;
+  background-color: #343434;
```
Comment (bohdan-degtiar) [inline frontend/src/libs/components/stepper/step-target/style.module.scss:31]: move `#343434` to variable

File: frontend/src/libs/components/stepper/step-target/stepper-target.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types';
```
Comment (what1s1ove) [inline frontend/src/libs/components/stepper/step-target/stepper-target.tsx:1]:
```suggestion
```

File: frontend/src/libs/components/stepper/step-target/style.module.scss [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  line-height: 16px;
+  letter-spacing: 0;
+  text-align: left;
+}
+
+input[type="checkbox"]::before {
+  margin: -4px 9px 0 0;
+}
+
+input[type="checkbox"]:checked::before {
```
Comment (what1s1ove) [inline frontend/src/libs/components/stepper/step-target/style.module.scss:23]: use classes
but better to use ReactSelect style api

File: frontend/src/libs/components/stepper/step-target/stepper-target.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+          ),
+        }}
+      />
+    </div>
+  );
+};
+
+TargetAudience.propTypes = {
+  className: PropTypes.string,
+};
```
Comment (what1s1ove) [inline frontend/src/libs/components/stepper/step-target/stepper-target.tsx:88]:
```suggestion
```
use typescript instead

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/stepper/step-target/stepper-target.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types';
+import { useForm } from 'react-hook-form';
```
Comment (bohdan-degtiar) [inline frontend/src/libs/components/stepper/step-target/stepper-target.tsx]: use `useAppForm`

File: frontend/src/libs/components/stepper/step-target/stepper-target.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                  checked={!!properties.isSelected}
+                  style={{
+                    marginRight: '9px',
+                  }}
+                />
+                <div>{properties.label}</div>
+              </div>
+            </components.Option>
+          ),
+        }}
```
Comment (bohdan-degtiar) [inline frontend/src/libs/components/stepper/step-target/stepper-target.tsx]: Don`t add customComponents this way.

File: frontend/src/libs/components/stepper/step-target/stepper-target.tsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      <Select
+        control={control}
+        name="ageGroup"
+        options={[
+          { value: 'kids', label: 'Kids 8-14' },
+          { value: 'teenager', label: 'Teenager 14-18' },
+          { value: 'young-adult', label: 'Young adult 18-35' },
+          { value: 'adult', label: 'Adult 35-65' },
+          { value: 'elderly', label: 'Elderly 65-99' },
+        ]}
```
Comment (bohdan-degtiar) [inline frontend/src/libs/components/stepper/step-target/stepper-target.tsx]: Move to separate variable.

**Review by bohdan-degtiar — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/stepper/step-target/stepper-target.tsx [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+        placeholder="Please select your industry"
+        errors={errors}
+        isMulti={true}
```
Comment (bohdan-degtiar) [inline frontend/src/libs/components/stepper/step-target/stepper-target.tsx]: Boolean property can set this way
```suggestion
        isMulti
```

**Comment by what1s1ove** [comment]
duplicate of https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-simpleland/pull/172

**Comment by what1s1ove** [comment]
duplicate of https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-simpleland/pull/172

**Pull request closed without merging**